### PR TITLE
move `transaction` module into `actions/` and rename to `set_transaction`

### PR DESCRIPTION
**Pull request by zachschuermann**
Prefactor for write-path work. this does three things
1. move transaction into actions module
2. rename module to `set_transaction` and `Transaction` to `SetTransaction`
3. add `commitInfo` to actions schema and define a commit info object which (for now) just includes kernel version

### CODE REVIEW

File: kernel/src/actions/mod.rs [hunk 1]
```diff
@@ -128,6 +131,11 @@ impl Protocol {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Schema)]
+pub struct CommitInfo {
+    pub kernel_version: Option<String>,
```
Comment (nicklan) [inline kernel/src/actions/mod.rs:136]: do we want to include the field that spark usually writes? maybe not until we have a use case.

File: kernel/src/actions/mod.rs [hunk 2]
```diff
@@ -16,23 +12,30 @@ use crate::actions::schemas::GetStructField;
...
+pub mod set_transaction;
+
+pub(crate) mod schemas;
+pub(crate) mod visitors;
+
 pub(crate) const ADD_NAME: &str = "add";
 pub(crate) const REMOVE_NAME: &str = "remove";
 pub(crate) const METADATA_NAME: &str = "metaData";
 pub(crate) const PROTOCOL_NAME: &str = "protocol";
 pub(crate) const TRANSACTION_NAME: &str = "txn";
```
Comment (nicklan) [inline kernel/src/actions/mod.rs:25]:
```suggestion
pub(crate) const SET_TRANSACTION_NAME: &str = "txn";
```

Will need to fix below too

File: kernel/src/actions/mod.rs [hunk 3]
```diff
@@ -16,23 +12,30 @@ use crate::actions::schemas::GetStructField;
...
+pub(crate) const COMMIT_INFO_NAME: &str = "commitInfo";
 
 static LOG_SCHEMA: LazyLock<StructType> = LazyLock::new(|| {
     StructType::new(vec![
         Option::<Add>::get_struct_field(ADD_NAME),
         Option::<Remove>::get_struct_field(REMOVE_NAME),
         Option::<Metadata>::get_struct_field(METADATA_NAME),
         Option::<Protocol>::get_struct_field(PROTOCOL_NAME),
-        Option::<Transaction>::get_struct_field(TRANSACTION_NAME),
+        Option::<SetTransaction>::get_struct_field(TRANSACTION_NAME),
```
Comment (nicklan) [inline kernel/src/actions/mod.rs:34]:
```suggestion
        Option::<SetTransaction>::get_struct_field(SET_TRANSACTION_NAME),
```

**Review by nicklan — CHANGES_REQUESTED** [review]
mostly lgtm, just a couple small nits

**Review by scottsand-db — APPROVED** [review]
LGTM. This brings the name of the "transaction" (`"txn"`) Delta protocol action to be en-par with how Delta-Kernel-API and Delta-Spark refer to it as in the code, a `SetTransaction`. That name is a signification improvement over "Transaction", and the unification between the kernels is also great.

File: kernel/src/actions/mod.rs (hunk 2 above)
Comment (zachschuermann) [reply kernel/src/actions/mod.rs:25]: great catch thanks

File: kernel/src/actions/mod.rs (hunk 1 above)
Comment (zachschuermann) [reply kernel/src/actions/mod.rs:136]: hm. I think there are _many_ fields spark usually writes and I didn't want to open that can of worms until we have a reason. thoughts?

Comment (nicklan) [reply kernel/src/actions/mod.rs:136]: Yep, that's fine, we can start small :)

File: kernel/src/actions/visitors.rs [hunk 4]
```diff
@@ -259,10 +260,10 @@ impl TransactionVisitor {
         row_index: usize,
         app_id: String,
         getters: &[&'a dyn GetData<'a>],
-    ) -> DeltaResult<Transaction> {
+    ) -> DeltaResult<SetTransaction> {
```
Comment (nicklan) [inline kernel/src/actions/visitors.rs:263]: (comment here because github sucks)

The struct should now be called `SetTransactionVisitor`

**Review by nicklan — CHANGES_REQUESTED** [review]
One last thing I noticed that needs to be renamed. You'll also need to fix the name in `inspect-table/src/main.rs` due to https://github.com/delta-incubator/delta-kernel-rs/pull/190

File: kernel/src/actions/visitors.rs (hunk 4 above)
Comment (zachschuermann) [reply kernel/src/actions/visitors.rs:281]: yep done, and did a few more renames just to prefix everything with `Set`

**Review by nicklan — APPROVED** [review]
lgtm!

**Review by scovich — APPROVED** [review]

**Pull request merged**